### DISCUSSION TITLE
How does tanstack query knows a refetched data is different than the stored data?

### DISCUSSION THREAD

**Discussion by sabres207** [TanStack/query#7866] [Q&A]
I'm new to tanstack query, using it in react.
I've noticed that when the website is refocused, it refetches the data.
I tried to print the data every time it changes, using useEffect, and I see it only changes when the data from backend actually changes.
Does it use deep comparison every time the page is refocused?

Also, is there an article to read about its' mechainsm?
Thanks!

**Comment by xeofd** [comment]
Queries have a `refetchOnWindowFocus` option, so any time the window is refocused it will refetch the data (I think even if not stale?). Can be turned off per query or as part of the QueryClient defaults you setup.

https://tanstack.com/query/latest/docs/framework/react/guides/window-focus-refetching

**Comment by TkDodo** [comment]
> Does it use deep comparison every time the page is refocused?

it uses structural sharing:
- https://tanstack.com/query/v5/docs/framework/react/guides/render-optimizations#structural-sharing
- https://tkdodo.eu/blog/react-query-render-optimizations#structural-sharing
